ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/SummaryCards/BySeveritySummaryCard.tsx

```diff
@@ -34,11 +41,12 @@ function BySeveritySummaryCard({
                 <Grid className="pf-u-pl-sm">
                     {severitiesCriticalToLow.map((severity) => {
                         const count = severityCounts[severity];
-                        const hasNoResults = count === 0;
-                        const isHidden = hiddenSeverities.has(severity);
+                        const hasNoResults = count.total === 0;
+                        const vulnSeverity = vulnCounterToSeverity[severity];
```

Comment:
Can you write a short comment (maybe above at `vulnCounterToSeverity` object) about the uses of the two severity keys?

Or, dare I ask, are the classic `'LOW_VULNERABILITY_SEVERITY'` keys used in any query payloads or responses? At least in this immediate context, the keys seem to be under frontend control:
* `vulnerabilitySeverityLabels`
* `SeverityIcons`

Reply:
Also, related, I found that we removed the `UNKNOWN_VULNERABILITY_SEVERITY` value from the front end type a while back. It looks like this is a valid response anywhere we could expect a severity from the BE though, and I've had a couple of cases of this in testing.

Do you see any reason to not add this back in? Possibly a discussion for the team meeting too.

Reply:
There was some light discussion in [this PR](https://github.com/stackrox/stackrox/pull/1328) regarding `UNKNOWN` but I didn't see anything beyond that. I think adding it back will result in very few compile errors, which we can solve by adding empty strings in the worst case.

As for duplications:

It looks like the `cve.proto.ts` type `VulnerabilitySeverity` is referenced 24 times, mostly by other types/utils, which are then imported primarily (if not entirely) in VulnMgmt 1.0, Vuln Reporting 1.0, and Risk Acceptance.

We also have `messages/common.ts` with a const `severityRatings` that is used in Policies. This value uses the severity shorthand version. `vulnSeverityLabels` in the same file maps the long version to the shorthand, which I wouldn't really think is a problem but might be a good canonical value->display mapper.

`severityColors.js` has _many_ different versions that map some severity to a color, I think mixing vuln severity with policy severity. It also contains a comment at the top `//  @TODO: Have one source of truth for severity colors` from 4 years ago 😄 

The new Workload CVEs intermixes the short style with the long style.

There are some .js files that use values like `MODERATE_VULNERABILITY_SEVERITY` and `Moderate` to refer to CVEs, but I'm less worried about those since they aren't typed and will either fall off with time, or become typed if they hang around.

There are also gql queries that reference resolvers like `low`, `moderate`, etc. that semantically seem like the same thing. We could alias those easily client side if we wanted, but we don't right now.

My general feeling is that "CVE Severity" is a concept that is used in enough places that it might be useful to keep it in a global `types` area. I'd be in favor of getting everything aligned with the current `VulnerabilitySeverity` since it is a direct mapping to the typed representation on the backend and would result in the least amount of moving code short-term. (Although too bad the API contract doesn't declare it, and we have to optimistically assume a `string` will be one of these values.)

***

Side note: Is `types/node.proto.ts` dead code? None of the types seem to be imported elsewhere. Transitively it looks like `types/vulnerability.proto.ts` might be dead code as well.